site/docs/patterns/forms.mdx

```diff
@@ -0,0 +1,839 @@
+            <ul>
+                <li>The user <em>should</em> be able to provide the right input based on the label and assistive text alone.</li>
+                <li>Assistive text <em>should</em> not be longer than the form control and more than two rows.</li>
+            </ul>
+        </li>
+    </ul>
+</div> 
+
+#### b) Tooltip
+A popup element that provides **context or background information** that is “nice to have” but not essential. 
+- *Should* be used to:
+    - Explain to users that may be unfamiliar with a particular form control.
+    - Offer a rationale for why certain information is needed.
+- An action button is added on the right side of the label.
+- The tooltip opens, when user clicks the action button.
+
+<Image src="../../static/patterns/form/form-guiding-tooltip@2x.png" alt="Tooltip" style="max-width:464px;" viewable /> 
+
+<div class="guideline-do" style="background: var(--color-success-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-do-label" style="color:var(--color-success); margin-bottom:var(--spacing-s);"><IconCheck style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Do</strong></div>
+    <ul style="margin-bottom:0;">
+        <li><strong>Be clear and concise.</strong>
+            <ul>
+                <li>Although tooltips can provide more information than the assistive text, they should preferably be less than two or three sentences long.</li>
+            </ul>
+        </li>
+        <li><strong>Provide a link to a separate page</strong>, if more detailed information is available.
+            <ul>
+                <li>Always open links from tooltips to a new window, so that the process of filling the form is not interrupted.</li>
+            </ul>
+        </li>
+    </ul>
+</div> 
+
+<div class="guideline-dont" style="background: var(--color-error-light); padding: var(--spacing-s); margin-bottom: var(--spacing-layout-2-xs);">
+    <div class="guideline-dont-label" style="color:var(--color-error); margin-bottom:var(--spacing-s);"><IconError style={{ marginRight: 'var(--spacing-3-xs)', verticalAlign: 'middle'}} size="s"/><strong>Don't</strong></div>
+    <ul style="margin-bottom:0;">
+        <li><strong>Make the tooltip open on hover.</strong>
+             <ul>
+                <li>The user should be able to select the text inside the tooltip element.</li>
+            </ul>
+        </li>
+        <li><strong>Tooltip must not be the sole way to provide essential information that the user must know to fill a value.</strong></li>
+    </ul>
+</div> 
+
+
+#### c) Placeholder text
+Placeholder text is a **supplementary visual aid** displayed inside the form control, that can be used form making the form easier to fill.
+
+It can be used to provide the user **a hint of what to enter or what is selected** with the form control: 
+- The preferred format of the input (e.g. Social security number: “DDMMYY-XXXX”)
```

Comment:
> The preferred format of the input (e.g. Social security number: “DDMMYY-XXXX”)

Itseäni hämää tämä esimerkki. Eikö tämä ole esimerkki juuri sellaisesta tiedosta, joka pitäisi olla assistive tekstissä? "Preferred" -sana tekee tästä sellaisen, että periaatteessa sopisi placeholderiin. Mutta sitten sotu kontekstina ei oikein, kun ei sitä voi syöttää missään muussa kuin tuossa muodossa.

Reply:
Muutin esimerkin muotoilua ja lisäsin alkuun E.g.: (for example Social security number: "E.g. 000000-123A”)